src/components/AllocationBar/AllocationBarSegment.tsx

```diff
@@ -0,0 +1,84 @@
+import { useSortable } from '@dnd-kit/sortable'
+import { CSS } from '@dnd-kit/utilities'
+import { AllocationBarDragHandle } from './AllocationBarDragHandle'
+import { AllocationBarResizeHandle } from './AllocationBarResizeHandle'
+import { AllocationItem } from './types'
+import { checkerboardStyle, MIN_SEGMENT_PERCENT } from './utils'
+
+// Formats a number to display at most 2 decimal places, but preserves all decimals if there are fewer than 2
+// Examples:
+// formatValue(10) => 10
+// formatValue(10.1) => 10.1
+// formatValue(10.12) => 10.12
+// formatValue(10.123) => 10.12
+// formatValue(10.126) => 10.13
+const formatValue = (val: number) => {
+  if (Number.isInteger(val)) return val
+  const decimals = val.toString().split('.')[1]?.length || 0
+  return val.toFixed(Math.min(2, decimals))
+}
```

Comment:
I would either return always string or number.
Or you could, you change for that is below. 
```suggestion
const formatValue = (val: number) => {
  return parseFloat(val.toFixed(2))
}
```